amazon/ionbenchmark/benchmark_runner.py

```diff
@@ -114,21 +114,51 @@ def test_fn():
         data_file = benchmark_spec.get_input_file()
 
         def test_fn():
-            with open(data_file, "rb") as f:
-                return loader_dumper.load(f)
+            format_option = benchmark_spec.get_format()
+            if _format.format_is_ion(format_option):
+                with open(data_file, "rb") as f:
+                    it = loader_dumper.load(f, parse_eagerly=False)
+                    while True:
+                        try:
+                            next(it)
+                        except StopIteration:
+                            break
+            elif _format.format_is_json(format_option):
+                with open(data_file, 'r') as f:
+                    while True:
+                        jsonl = f.readline()
+                        if jsonl == '':
+                            break
+                        loader_dumper.loads(jsonl)
+            elif _format.format_is_cbor(format_option):
+                with open(data_file, 'br') as f:
+                    while True:
+                        try:
+                            loader_dumper.load(f)
+                        except EOFError:
+                            break
+            elif _format.format_is_protobuf(format_option):
+                with open(data_file, "rb") as f:
+                    return loader_dumper.load(f)
 
     elif match_arg == ['file', 'write', 'load_dump']:
+        # This method should return a list that holds all top_level values.
         data_obj = benchmark_spec.get_data_object()
         data_format = benchmark_spec.get_format()
-        if _format.format_is_binary(data_format) or _format.format_is_ion(data_format):
+        if _format.format_is_protobuf(data_format):
             def test_fn():
                 with tempfile.TemporaryFile(mode="wb") as f:
                     return loader_dumper.dump(data_obj, f)
+        elif _format.format_is_binary(data_format) or _format.format_is_ion(data_format):
+            def test_fn():
+                with tempfile.TemporaryFile(mode="ab") as f:
+                    for top_level_obj in data_obj:
+                        loader_dumper.dump(top_level_obj, f)
```

Comment:
For Ion, is this going to write a new symbol table for each top-level value? That would not lead to the most efficient encoding. How does the performance and data size for Ion compare when using this technique vs using a single dump of the entire list of top-level values?

Reply:
```
loader_dumper.dump(data_obj, f, sequence_as_stream=True)
| (ion_binary,dumps,service_log_legacy) |       22263038 |  2027258750.00 |   2035399545.00 |               42571303 |
```
```
loader_dumper.dump(data_obj, f, sequence_as_stream=False)
| (ion_binary,dumps,service_log_legacy) |       22263038 |  2010238558.40 |   2024798020.82 |               42571457 |
```
```
for top_level_obj in data_obj:
    loader_dumper.dump(top_level_obj, f)
| (ion_binary,dumps,service_log_legacy) |       22263038 |  2486109233.40 |   2495525362.50 |                  44581 |
```

Changed to `loader_dumper.dump(data_obj, f, sequence_as_stream=True)`